st2api/st2api/controllers/v1/actions.py

```diff
@@ -264,6 +264,8 @@ def delete(self, ref_or_id, requester_user):
                 "Exception was %s",
                 e,
             )
+            action_db.id = None
+            Action.add_or_update(action_db)
             abort(http_client.INTERNAL_SERVER_ERROR, six.text_type(e))
```

Comment:
Also, if the deletion is related to permission error, can you change the error code to 403 Forbidden here? Otherwise for any other error, return 500 is fine.